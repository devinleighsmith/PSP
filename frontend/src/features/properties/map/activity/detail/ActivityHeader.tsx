import { StyledDivider } from 'components/common/styles';
import { UserNameTooltip } from 'components/common/UserNameTooltip';
import { HeaderField } from 'features/mapSideBar/tabs/HeaderField';
import { Api_File } from 'models/api/File';
import * as React from 'react';
import { Col, Row } from 'react-bootstrap';
import { prettyFormatDate } from 'utils';

export interface IActivityHeaderProps {
  file: Api_File;
}

export const ActivityHeader: React.FunctionComponent<IActivityHeaderProps> = ({ file }) => {
  const leftColumnWidth = '5';
  const leftColumnLabel = '3';
  return (
    <Row className="no-gutters pl-4 pr-4">
      <Col>
        <Row className="no-gutters">
          <Col xs={leftColumnWidth}>
            <HeaderField label="File: " labelWidth={leftColumnLabel}>
              {file?.fileNumber}
            </HeaderField>
          </Col>
          <Col className="text-right">
            Created: <strong>{prettyFormatDate(file?.appCreateTimestamp)}</strong> by{' '}
            <UserNameTooltip userName={file?.appCreateUserid} userGuid={file?.appCreateUserGuid} />
          </Col>
        </Row>
        <Row className="no-gutters">
          <Col xs={leftColumnWidth}></Col>
          <Col className="text-right">
            Last updated: <strong>{prettyFormatDate(file?.appLastUpdateTimestamp)}</strong> by{' '}
            <UserNameTooltip
              userName={file?.appLastUpdateUserid}
              userGuid={file?.appLastUpdateUserGuid}
            />
          </Col>
        </Row>
        <Row className="no-gutters">
          <Col xs={leftColumnWidth}></Col>
          <Col>
            <HeaderField className="justify-content-end" label="Status:">
              {file?.fileStatusTypeCode?.description}
            </HeaderField>
          </Col>
        </Row>
        <StyledDivider />
      </Col>
    </Row>
  );
};

export default ActivityHeader;
