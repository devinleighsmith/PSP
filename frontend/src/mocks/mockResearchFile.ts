export const getMockResearchFile = () => ({
  roadName: 'Bubba Loo Lane',
  roadAlias: 'Bubba Loo Expressway',
  researchProperties: [
    {
      id: 1,
      propertyName: 'Bubba 1',
      isDisabled: false,
      property: {
        id: 1,
        anomalies: [],
        tenures: [],
        roadTypes: [],
        adjacentLands: [],
        region: {
          id: 1,
          description: 'South Coast Region',
          isDisabled: false,
        },
        district: {
          id: 1,
          description: 'Lower Mainland District',
          isDisabled: false,
        },
        dataSourceEffectiveDate: '2021-08-31T00:00:00',
        latitude: 54.794202998379006,
        longitude: -127.18413347053901,
        isSensitive: false,
        isRwyBeltDomPatent: false,
        pid: 7723385,
        pin: 90069930,
        landArea: 1,
        isVolumetricParcel: false,
        location: {
          coordinate: {
            x: -127.18413347053901,
            y: 54.794202998379006,
          },
        },
        rowVersion: 3,
      },
      purposeTypes: [
        {
          id: 1,
          propertyPurposeType: {
            id: 'ACQUIRE',
            description: 'Acquisition',
            isDisabled: false,
          },
          propertyResearchFileId: 1,
          rowVersion: 1,
        },
      ],
      rowVersion: 1,
    },
  ],
  requestDate: '2022-08-02T00:00:00',
  requestSourceType: {
    id: 'HQ',
    description: 'Headquarters (HQ)',
    isDisabled: false,
  },
  researchFilePurposes: [
    {
      id: '1',
      researchPurposeTypeCode: {
        id: 'ACQUIRE',
        description: 'Acquisition',
        isDisabled: false,
      },
      appCreateTimestamp: '2022-08-25T16:23:17.943',
      appLastUpdateTimestamp: '2022-08-25T16:23:17.943',
      appLastUpdateUserid: 'dbo',
      appCreateUserid: 'dbo',
      rowVersion: 1,
    },
  ],
  id: 1,
  fileName: 'Bubba BBQ Whirled',
  fileNumber: 'R-Bubba1',
  fileStatusTypeCode: {
    id: 'ACTIVE',
    description: 'Active',
    isDisabled: false,
  },
  appCreateTimestamp: '2022-08-27T16:23:17.863',
  appLastUpdateTimestamp: '2022-08-25T16:23:17.863',
  appLastUpdateUserid: 'dbo',
  appCreateUserid: 'dbo',
  rowVersion: 1,
});
