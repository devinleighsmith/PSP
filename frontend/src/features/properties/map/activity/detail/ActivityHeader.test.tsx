import axios from 'axios';
import MockAdapter from 'axios-mock-adapter';
import { mockAcquisitionFileResponse } from 'mocks/mockAcquisitionFiles';
import { getMockResearchFile } from 'mocks/mockResearchFile';
import { prettyFormatDate } from 'utils';
import { render, RenderOptions } from 'utils/test-utils';

import ActivityHeader, { IActivityHeaderProps } from './ActivityHeader';

const mockAxios = new MockAdapter(axios);

describe('ActivityHeader component', () => {
  // render component under test
  const setup = (props: IActivityHeaderProps, renderOptions: RenderOptions = {}) => {
    const utils = render(<ActivityHeader file={props.file} />, {
      ...renderOptions,
    });

    return { ...utils };
  };

  beforeEach(() => {
    mockAxios.onGet(new RegExp('users/info/*')).reply(200, {});
  });

  afterEach(() => {
    mockAxios.reset();
    jest.clearAllMocks();
  });

  it('renders as expected', () => {
    const { asFragment } = setup({ file: getMockResearchFile() });
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders as expected when an acquisition file is provided', async () => {
    const testAcquisitionFile = { ...mockAcquisitionFileResponse(), id: 1 };
    const { getByText } = setup({ file: { ...mockAcquisitionFileResponse(), id: 1 } });

    expect(getByText(testAcquisitionFile.fileNumber as string)).toBeVisible();
    expect(getByText(prettyFormatDate(testAcquisitionFile.appCreateTimestamp))).toBeVisible();
    expect(getByText(prettyFormatDate(testAcquisitionFile.appLastUpdateTimestamp))).toBeVisible();
    expect(getByText(testAcquisitionFile?.fileStatusTypeCode?.description ?? '')).toBeVisible();
  });

  it('renders as expected when a research file is provided', async () => {
    const testResearchFile = getMockResearchFile();
    const { getByText } = setup({ file: testResearchFile });

    expect(getByText(testResearchFile.fileNumber as string)).toBeVisible();
    expect(getByText(prettyFormatDate(testResearchFile.appCreateTimestamp))).toBeVisible();
    expect(getByText(prettyFormatDate(testResearchFile.appLastUpdateTimestamp))).toBeVisible();
    expect(getByText(testResearchFile.fileStatusTypeCode.description)).toBeVisible();
  });
});
