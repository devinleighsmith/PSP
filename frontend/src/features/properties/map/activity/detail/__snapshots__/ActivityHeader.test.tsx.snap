// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ActivityHeader component renders as expected 1`] = `
<DocumentFragment>
  <div
    class="Toastify"
  />
  <div
    class="container"
  />
  <div
    class="Toastify"
  />
  .c1 {
  margin-top: 0.5rem;
  margin-bottom: 1.5rem;
  border-bottom-style: solid;
  border-bottom-color: grey;
  border-bottom-width: 0.1rem;
}

.c0 {
  font-weight: bold;
}

<div
    class="no-gutters pl-4 pr-4 row"
  >
    <div
      class="col"
    >
      <div
        class="no-gutters row"
      >
        <div
          class="col-5"
        >
          <div
            class="row"
          >
            <div
              class="pr-0 text-left col-3"
            >
              <label>
                File: 
              </label>
            </div>
            <div
              class="pl-1 text-left col-auto"
            >
              <strong>
                R-Bubba1
              </strong>
            </div>
          </div>
        </div>
        <div
          class="text-right col"
        >
          Created: 
          <strong>
            Aug 27, 2022
          </strong>
           by 
          <span
            class="tooltip-icon"
            data-testid="tooltip-icon"
            id="userNameTooltip"
          >
            <span
              class="c0"
            >
              dbo
            </span>
          </span>
        </div>
      </div>
      <div
        class="no-gutters row"
      >
        <div
          class="col-5"
        />
        <div
          class="text-right col"
        >
          Last updated: 
          <strong>
            Aug 25, 2022
          </strong>
           by 
          <span
            class="tooltip-icon"
            data-testid="tooltip-icon"
            id="userNameTooltip"
          >
            <span
              class="c0"
            >
              dbo
            </span>
          </span>
        </div>
      </div>
      <div
        class="no-gutters row"
      >
        <div
          class="col-5"
        />
        <div
          class="col"
        >
          <div
            class="justify-content-end row"
          >
            <div
              class="pr-0 text-left col-auto"
            >
              <label>
                Status:
              </label>
            </div>
            <div
              class="pl-1 text-left col-auto"
            >
              <strong>
                Active
              </strong>
            </div>
          </div>
        </div>
      </div>
      <div
        class="c1"
      />
    </div>
  </div>
</DocumentFragment>
`;
